minidump-processor/src/stackwalker/amd64.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2015 Ted Mielczarek. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+
+// Note since x86 and Amd64 have basically the same ABI, this implementation
+// is written to largely erase the details of the two wherever possible,
+// so that it can be copied between the two with minimal changes. It's not
+// worth the effort to *actually* unify the implementations.
+
+use crate::process_state::{FrameTrust, StackFrame};
+use crate::stackwalker::unwind::Unwind;
+use minidump::format::CONTEXT_AMD64;
+use minidump::{
+    MinidumpContext, MinidumpContextValidity, MinidumpMemory, MinidumpModuleList,
+    MinidumpRawContext,
+};
+use std::collections::HashSet;
+
+type Pointer = u64;
+const POINTER_WIDTH: Pointer = 8;
+const INSTRUCTION_REGISTER: &str = "rip";
+const STACK_POINTER_REGISTER: &str = "rsp";
+const FRAME_POINTER_REGISTER: &str = "rbp";
+
+fn get_caller_by_frame_pointer(
+    ctx: &CONTEXT_AMD64,
+    valid: &MinidumpContextValidity,
+    _trust: FrameTrust,
+    stack_memory: &MinidumpMemory,
+    _modules: &MinidumpModuleList,
+) -> Option<StackFrame> {
+    match valid {
```

Comment:
I think this is marginally simpler as `if let MinidumpContextValidity::Some(ref which) => { … }`.